TrafficCapture/captureKafkaOffloader/src/main/java/org/opensearch/migrations/trafficcapture/kafkaoffloader/KafkaCaptureFactory.java

```diff
@@ -70,7 +60,9 @@ public IChannelConnectionCaptureSerializer createOffloader(String connectionId)
                     CompletableFuture cf = new CompletableFuture<>();
                     // Async request to Kafka cluster
                     producer.send(record, handleProducerRecordSent(cf, recordId));
-                    return cf;
+                    // A more desirable way to cut off our tree of cf aggregation should be investigated
+                    aggregateCf.set(aggregateCf.get().isDone() ? cf : CompletableFuture.allOf(aggregateCf.get(), cf));
```

Comment:
The order of messages processed by Kafka.  Each one will fire independent of previous ones that were fired.  We'll only have the last close() call be aware of when all of the pieces have been committed (in some order).

Reply:
Gotcha yes the Kafka messages are firing independent of each other but our CompletableFuture is synchronously expanding so that the final close() will be a future comprised of the previous ones